### PR TITLE
Notification Preferences ControlPanel rebased

### PR DESCRIPTION
**Pull request by jeffersondarcy**
Notification Preferences
System and Marketing Tabs

replaces https://github.com/corona-school/user-app/pull/31

### CODE REVIEW

File: src/hooks/useNotificationPreferences.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const notificationPreferencesQuery = gql`
+  query {
+    me {
+      notificationPreferences
+    }
+  }
+`
+
+const notificationPreferencesMutation = gql`
```
Comment (LomyW) [inline src/hooks/useNotificationPreferences.ts:13]: maybe renaming to `updateNotificationPreferences`

File: src/hooks/useNotificationPreferences.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+const useUserPreferences = () => {
+  const [userPreferences, setUserPreferencesPrivate] =
+    useState<NotificationPreferences>({})
+
+  const { data, loading, error } = useQuery(notificationPreferencesQuery)
+
+  const [updateUserPreferences] = useMutation(notificationPreferencesMutation)
+
+  const updateUserPreference = (category: string, channel: string, value: boolean) => {
+    const preferences = { ...userPreferences, [category]: {[channel]: value}}
```
Comment (LomyW) [inline src/hooks/useNotificationPreferences.ts:28]: nice, is easier to read now 😄

**Review by LomyW — APPROVED** [review]
looks good! 🥳

File: src/pages/notification/NotficationControlPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        lg: false,
+    });
+
+    const width = useBreakpointValue({
+        base: '90%',
+        lg: '90%',
+    });
+
+    return (
+        <NotificationPreferencesContext.Provider value={{ userPreferences, updateUserPreference }}>
```
Comment (timidak) [inline src/pages/notification/NotficationControlPanel.tsx:29]: Nice approach! Using a context here looks pretty good! 🎉

**Review by timidak — APPROVED** [review]
LGTM 🎉

**Pull request merged**